app/client/src/apiService.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable no-console */
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { Commit, ActionContext } from 'vuex';
+import { ResponseSuccess, ResponseFailure, BeebopError } from './types';
+
+export interface ResponseWithType<T> extends ResponseSuccess {
+    data: T
+}
+
+export function isAPIError(object: any): object is BeebopError {
+  return typeof object.error === 'string'
```

Comment:
do you need `object &&` here as for `isAPIResponseFailure`?

Reply:
Yes, I think that makes sense so it does not return an error in case the object is `null`